### PR TITLE
Mark bytes as consumed 

### PR DESCRIPTION
**Pull request by Tratcher**
#13372 @Pilchie 
This looks like a preview9 regression from #12749 where we refactored the form parser. It's externally reported as impacting OpenIdConnect auth, but I was only able to reproduce it using WsFed auth. The request fails and the client is unable to log in. I don't see any immediate workarounds.

The issue happens if the form is long enough to take the slow path (multiple buffer segments), but the last key=value pair is short and takes the single segment fast path. In that case it fails to mark the last bytes as consumed and the parser mistakenly aborts.

Compare the fix to what happens in the slow path on lines 294 and 295 bellow.

### CODE REVIEW

**Comment by dougbu** [comment]
This looks ready but we need an approver (beyond QB-approval, which this has)

**Comment by mmitche** [comment]
Can we get this reviewed and checked in so we can get a build for Sunday night validation?

File: src/Http/WebUtilities/test/FormPipeReaderTests.cs
Comment (BrennanConroy) [inline src/Http/WebUtilities/test/FormPipeReaderTests.cs]: I'm assuming this would have failed before the fix?

**Review by BrennanConroy — APPROVED** [review]

**Review by jkotalik — APPROVED** [review]

File: src/Http/WebUtilities/test/FormPipeReaderTests.cs
Comment (Tratcher) [reply src/Http/WebUtilities/test/FormPipeReaderTests.cs]: Right, this covers the repro scenario.

**Comment by Tratcher** [comment]
@JunTaoLuo @Pilchie admin merge please.

**Pull request merged**
